File: Algorithms_medium/49_Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        chars = defaultdict(list)
        for s in strs:
            k = ''.join(sorted(list(s)))
            chars[k].append(s)
        return list(chars.values())
'''
如题，做法是哈希表的思路，建立一个字典，将不同字符串组合sorted重排列后，作为字典的key
然后将不同的排列形式存入key下，最后返回字典的value
'''

File: Algorithms_C#/49_Group_Anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        IDictionary<string, IList<string>> dictionary = new Dictionary<string, IList<string>>();
        foreach (string str in strs){
            char[] array = str.ToCharArray();
            Array.Sort(array);
            string key = new string(array);
            dictionary.TryAdd(key, new List<string>());
            dictionary[key].Add(str);
        }
        return new List<IList<string>>(dictionary.Values);
    }
}